File: add_favicon.js
const fs = require('fs');

const files = fs.readdirSync('.', { withFileTypes: true })
    .filter(dirent => !dirent.isDirectory() && dirent.name.endsWith('.html'))
    .map(dirent => dirent.name);

for (const file of files) {
    let content = fs.readFileSync(file, 'utf8');
    if (!content.includes('<link rel="icon"')) {
        content = content.replace('<head>', '<head>\n    <!-- Favicon -->\n    <link rel="icon" type="image/jpeg" href="Logo.jpeg">');
        fs.writeFileSync(file, content);
        console.log(`Added favicon to ${file}`);
    }
}

if (fs.existsSync('./admin')) {
    const adminFiles = fs.readdirSync('./admin', { withFileTypes: true })
        .filter(dirent => !dirent.isDirectory() && dirent.name.endsWith('.html'))
        .map(dirent => `./admin/${dirent.name}`);

    for (const file of adminFiles) {
        let content = fs.readFileSync(file, 'utf8');
        if (!content.includes('<link rel="icon"')) {
            content = content.replace('<head>', '<head>\n    <!-- Favicon -->\n    <link rel="icon" type="image/jpeg" href="../Logo.jpeg">');
            fs.writeFileSync(file, content);
            console.log(`Added favicon to ${file}`);
        }
    }
}

console.log("Favicon injection complete!");

File: add_favicon.py
import os
import glob

# Process root HTML files
files = glob.glob(r'd:\Peptide\*.html')
for f in files:
    with open(f, 'r', encoding='utf-8') as file:
        content = file.read()
    
    # Check if favicon already exists
    if '<link rel="icon"' not in content:
        content = content.replace('<head>', '<head>\n    <!-- Favicon -->\n    <link rel="icon" type="image/jpeg" href="Logo.jpeg">', 1)
        with open(f, 'w', encoding='utf-8') as file:
            file.write(content)
            print(f"Added favicon to {f}")

# Process admin HTML files (requires ../ path)
admin_files = glob.glob(r'd:\Peptide\admin\*.html')
for f in admin_files:
    with open(f, 'r', encoding='utf-8') as file:
        content = file.read()
    
    if '<link rel="icon"' not in content:
        content = content.replace('<head>', '<head>\n    <!-- Favicon -->\n    <link rel="icon" type="image/jpeg" href="../Logo.jpeg">', 1)
        with open(f, 'w', encoding='utf-8') as file:
            file.write(content)
            print(f"Added favicon to {f}")

print("Favicon injection complete!")
